star-ros2/src/star_safety_monitor/test/test_safety_monitor.cpp

```
@@ -0,0 +1,214 @@
+  odom_pub->publish(odom_msg);
+  executor.spin_some(SPIN_TIMEOUT);
+
+  // Give the node time to process
+  std::this_thread::sleep_for(LIFECYCLE_TRANSITION_DELAY);
+  executor.spin_some(SPIN_TIMEOUT);
+
+  // Test passes if no exceptions are thrown
+  EXPECT_TRUE(true);
+}
+
+TEST_F(SafetyMonitorTest, BatterySafetyChecks)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)